### ISSUE TITLE
Pylint can't find type alias defined inside a class definition.

### ISSUE THREAD

**Issue by asincero** [PyCQA/pylint#1766]
### Steps to reproduce

Pylint can't find a type alias defined inside a class definition.  For example:

```python
import typing

class MyClass:
    Things = typing.Sequence[str]

    def use_things(self, o: Things) -> None:
        pass

    def get_things(self) -> Things:
        pass
```

### Current behavior

Pylint reports "Things" on line 9 as being an undefined variable.

### Expected behavior

No warnings or errors for the above code.

### pylint --version output

pylint 1.7.4,
astroid 1.5.3
Python 3.6.3 (default, Oct  4 2017, 06:09:15)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]

**Comment by dstromberg** [comment]
Not sure if it's a bug or not, but if you define a class that uses itself, and is typechecked via mypy, the documented behavior of mypy is that you need to put the class name in quotes because the class doesn't exist until it's finished being defined.

**Comment by PCManticore** [comment]
This is definitely a bug.

**Comment by DanielNoord** [comment]
I can't reproduce this on `main`. Therefore I am closing this.

Please report back if the issue still persists!
